File: analyzers/its/sources/ManuallyAddedNoncompliantIssues.VB/IntentionalFindings/AvoidDateTimeNowForBenchmarking.vb
Imports System

Friend Class AvoidDateTimeNowForBenchmarking
    Public Sub S6561()
        Dim start = Date.Now
        MethodToBeBenchmarked()
        Console.WriteLine($"{(Date.Now - start).TotalMilliseconds} ms") ' Noncompliant (S6561)
    End Sub
    Private Sub MethodToBeBenchmarked()
    End Sub
End Class

File: analyzers/its/sources/ManuallyAddedNoncompliantIssues.CS/NetFramework48/AvoidDateTimeNowForBenchmarking.cs
using System;

namespace NetFramework48
{
    internal class AvoidDateTimeNowForBenchmarking
    {
        public void S6561()
        {
            var start = DateTime.Now;
            MethodToBeBenchmarked();
            Console.WriteLine($"{(DateTime.Now - start).TotalMilliseconds} ms");
        }

        void MethodToBeBenchmarked() { }
    }
}
